meson: build detection issues on OpenBSD

Meson is not able to find iconv. Looking at the Meson log it does not appear to be utilizing the pkg-config file.

```
Run-time dependency iconv found: NO (tried builtin and system)

../../pobj/mpv-0.35.0/mpv-0.35.0/meson.build:605:0: ERROR: Dependency "iconv" not found, tried builtin and system
```

```
Running compile:
Working directory:  /home/ports/pobj/mpv-0.35.0/build-amd64/meson-private/tmp0arqvwug
Command line:  cc /home/ports/pobj/mpv-0.35.0/build-amd64/meson-private/tmp0arqvwug/testfile.c -o /home/ports/pobj/mpv-0.35.0/build-amd64/meson-private/tmp0arqvwug/output.exe -O2 -pipe -g -D_FILE_OFFSET_BITS=64 -O0 -Werror=implicit-function-declaration

Code:
 #include <iconv.h>

int main() {
    iconv_open("","");
}
Compiler stdout:

Compiler stderr:
 /home/ports/pobj/mpv-0.35.0/build-amd64/meson-private/tmp0arqvwug/testfile.c:1:10: fatal error: 'iconv.h' file not found
#include <iconv.h>
         ^~~~~~~~~
1 error generated.

Running compile:
Working directory:  /home/ports/pobj/mpv-0.35.0/build-amd64/meson-private/tmpe2s2p9ga
Command line:  cc /home/ports/pobj/mpv-0.35.0/build-amd64/meson-private/tmpe2s2p9ga/testfile.c -E -P -O2 -pipe -g -D_FILE_OFFSET_BITS=64 -P -O0 -Werror=implicit-function-declaration

Code:

        #ifdef __has_include
         #if !__has_include("iconv.h")
          #error "Header 'iconv.h' could not be found"
         #endif
        #else
         #include <iconv.h>
        #endif
Compiler stdout:


Compiler stderr:
 /home/ports/pobj/mpv-0.35.0/build-amd64/meson-private/tmpe2s2p9ga/testfile.c:4:12: error: "Header 'iconv.h' could not be found"
          #error "Header 'iconv.h' could not be found"
           ^
1 error generated.

Running compile:
Working directory:  /home/ports/pobj/mpv-0.35.0/build-amd64/meson-private/tmpfau0706l
Command line:  cc /home/ports/pobj/mpv-0.35.0/build-amd64/meson-private/tmpfau0706l/testfile.c -o /home/ports/pobj/mpv-0.35.0/build-amd64/meson-private/tmpfau0706l/output.exe -O2 -pipe -g -D_FILE_OFFSET_BITS=64 -O0 -Werror=implicit-function-declaration -Wl,--start-group -liconv -Wl,--end-group -Wl,--allow-shlib
-undefined

Code:
 int main(void) { return 0; }

Compiler stdout:

Compiler stderr:
 ld: error: unable to find library -liconv
cc: error: linker command failed with exit code 1 (use -v to see invocation)

Run-time dependency iconv found: NO (tried builtin and system)

../../pobj/mpv-0.35.0/mpv-0.35.0/meson.build:605:0: ERROR: Dependency "iconv" not found, tried builtin and system
```

```
../../pobj/mpv-0.35.0/mpv-0.35.0/meson.build:719:0: ERROR: C shared or static library 'atomic' not found
```

OpenBSD and FreeBSD do not use libatomic.

--- a/meson.build
+++ b/meson.build
@@ -716,10 +716,11 @@ if features['sdl2-gamepad']
     sources += files('input/sdl_gamepad.c')
 endif
 
-stdatomic = cc.find_library('atomic', required: get_option('stdatomic'))
-features += {'stdatomic': stdatomic.found()}
+stdatomic_dep = cc.find_library('atomic', required: false)
+features += {'stdatomic': cc.has_header_symbol('stdatomic.h', 'atomic_int', dependencies: stdatomic_dep,
+                                               required: get_option('stdatomic'))}
 if features['stdatomic']
-    dependencies += stdatomic
+    dependencies += stdatomic_dep
 endif
 
 uchardet_opt = get_option('uchardet').require(

